plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/server/LocalAppEngineServerDelegate.java

```diff
@@ -80,6 +103,50 @@ public IStatus canModifyModules(IModule[] add, IModule[] remove) {
     return Status.OK_STATUS;
   }
 
+  /**
+   * Check that the associated projects have unique App Engine Service IDs.
+   */
+  @VisibleForTesting
+  IStatus checkConflictingServiceIds(IModule[] current, IModule[] toBeAdded,
+      IModule[] toBeRemoved, Function<IModule, String> serviceIdFunction) {
```

Comment:
It was the easiest way simplify testing.  Or else I'd have to figure out some way to intercept `ModuleUtil#getServiceId()`.